prow/cmd/deck/static/prow/prow.ts

```diff
@@ -673,24 +667,25 @@ function createRerunCell(modal: HTMLElement, rerunElement: HTMLElement, prowjob:
         const runButton = document.createElement('a');
         runButton.innerHTML = "<button class='mdl-button mdl-js-button'>Rerun</button>";
         if (login === "") {
-            runButton.href = `/github-login?dest=%2F?rerun=work_in_progress`;
+            runButton.href = `/github-login?dest=%2F?rerun=gh_redirect`;
         } else {
             if (rerunCreatesJob) {
-                runButton.onclick = () => {
-                    const form = document.createElement('form');
-                    form.method = 'POST';
-                    form.action = `${url}`;
-                    const tokenInput = document.createElement('input');
-                    tokenInput.type = 'hidden';
-                    tokenInput.name = 'gorilla.csrf.Token';
-                    tokenInput.value = csrfToken;
-                    form.append(tokenInput);
-                    c.appendChild(form);
-                    form.submit();
+                runButton.onclick = async () => {
+                    const result = await fetch(url, {
+                        headers: {
+                            "Content-type": "application/x-www-form-urlencoded; charset=UTF-8",
+                            "X-CSRF-Token": csrfToken,
+                        },
+                        method: 'post',
+                    });
+                    const data = await result.text();
+                    rerunElement.innerHTML = data;
                 };
             } else {
-                runButton.href = `/?rerun=work_in_progress`;
-                runButton.onclick = () => gtag("event", "troll_rerun_popup", {event_category: "engagement", transport_type: "beacon"});
+                runButton.onclick = () => {
+                    gtag("event", "troll_rerun_popup", {event_category: "engagement", transport_type: "beacon"});
+                    rerunElement.innerHTML = "Nice try! The direct rerun feature hasn't been implemented yet, so that button does nothing.";
```

Comment:
Instead of this, can we just remove the button? having a useless button that the instance has opted out of doesn't seem useful.

Reply:
Done. Not sure why I didn't think of this earlier